solver/src/solver/naive_solver/multi_order_solver.rs

```diff
@@ -77,7 +78,7 @@ pub fn solve(
 /// for that pair is not available.
 ///
 fn solve_orders(
-    orders: impl Iterator<Item = LimitOrder> + Clone,
+    orders: &[LimitOrder],
```

Comment:
Love this! Is this something we could have always done? Or do some of the changes made here make it possible?

Reply:
No, we could have always done it. I did it mostly because I needed to iterate over `orders` twice and didn't want to clone a whole `Vec` to do it.